### PR TITLE
Complete O_flags used for file open

### PR DESCRIPTION
**Pull request by BCSharp**
Includes part of the solution for #1225.

### CODE REVIEW

File: Src/IronPython/Modules/_io.cs [hunk 1]
```diff
@@ -54,6 +55,19 @@ public static partial class PythonIOModule {
 
         private static int O_EXCL => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? 0x400 : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? 0x800 : 0x80;
 
+        [PythonHidden(PlatformsAttribute.PlatformFamily.Windows)]
```
Comment (slozier) [inline Src/IronPython/Modules/_io.cs:58]: Having `PythonHidden` on a `private` seems kind of odd but I guess it doesn't hurt...

File: Tests/test_file.py [hunk 2]
```diff
@@ -701,7 +701,7 @@ def test_errors(self):
 
         with self.assertRaises(OSError) as cm:
             open('path_too_long' * 100)
-        self.assertEqual(cm.exception.errno, (36 if is_posix else 22) if is_netcoreapp and not is_posix or sys.version_info >= (3,6) else 2)
+        self.assertEqual(cm.exception.errno, (63 if is_osx else 36 if is_linux else 22) if is_netcoreapp and not is_posix or sys.version_info >= (3,6) else 2)
```
Comment (slozier) [inline Tests/test_file.py:704]: Not that I mind the explicit number check, but wondering if we should be using the `errno` module values instead to make things easier to decipher.

**Review by slozier — APPROVED** [review]
Looks good to me!

File: Tests/test_file.py (hunk 2 above)
Comment (BCSharp) [reply Tests/test_file.py:704]: Good point, assuming that `errno` is not buggy 😄

File: Src/IronPython/Modules/_io.cs (hunk 1 above)
Comment (BCSharp) [reply Src/IronPython/Modules/_io.cs:58]: Yes, it doesn't hurt, but the less verbosity the better. I assume the same applies to `internal`, though I see there are a few `PythonHidden` that are `internal` already in the codebase.

Comment (slozier) [reply Src/IronPython/Modules/_io.cs]: Weird, I wonder why we'd have them on `internal` members. Maybe something that was `public` at some point?

Comment (BCSharp) [reply Src/IronPython/Modules/_io.cs]: Look for example at `ByteArray.UnsafeByteList`. It was established 5 years ago as `internal`, but 4 years ago you added `PythonHidden` in #1000. Do you remember why?

Comment (slozier) [reply Src/IronPython/Modules/_io.cs]: Can't say I remember, but looking at https://github.com/IronLanguages/ironpython3/pull/693 I changed `Bytes.GetUnsafeByteArray` (which was public) to a `public` property and then in a follow up commit made it `internal`. I guess I forgot to remove the `PythonHidden` when I did this. I'm guessing `ByteArray.UnsafeByteList` is a copy/paste...

**Pull request merged**